File: ESCAlertController/ESCAlertController/AlertViewController/ESCAlertAction.m
//
//  ESCAlertAction.m
//  ESCAlertController
//
//  Created by xiatian on 2023/6/8.
//

#import "ESCAlertAction.h"

@interface ESCAlertAction ()


@property(nonatomic,copy,readwrite)NSString* title;

@property(nonatomic,copy,readwrite)UIColor* titleColor;

@property(nonatomic,copy,readwrite)void((^handler)(ESCAlertAction *action));

@end


@implementation ESCAlertAction

+ (instancetype)actionWithTitle:( NSString *)title titleColor:(UIColor *)titleColor handler:(void (^ )(ESCAlertAction *action))handler {
    ESCAlertAction *action = [[ESCAlertAction alloc] init];
    action.title = title;
    action.titleColor = titleColor;
    [action setHandler:^(ESCAlertAction *action) {
        if (handler) {
            handler(action);
        }
    }];
    return action;;
}



@end

File: ESCAlertController/Swift_ESCAlertController/AlertViewController/ESCAlertAction.swift
//
//  ESCAlertAction.swift
//  Swift_ESCAlertController
//
//  Created by xiatian on 2023/6/8.
//

import UIKit

class ESCAlertAction: NSObject {
    
    var title:String?
    
    var titleColor:UIColor?
    
    var handler:((_ alertAction:ESCAlertAction)->Void)?
    
    public init(title: String? = nil, titleColor: UIColor? = nil,handler:@escaping ((_ alertAction:ESCAlertAction)->Void)) {
        self.title = title
        self.titleColor = titleColor
        self.handler = handler
    }
    


}
